File: tests/expected/sys_argv.kt


fun main(argv: Array<String>) {
    var a: Array<String> = (arrayOf("") + argv)
    var cmd: String = a[0]
    if (cmd == "dart") {
// pass
    } else {
        assert("sys_argv" in cmd)
    }
    if (a.size > 1) {
        if (true) {
            val __tmp1 = a[1]
            println("$__tmp1")
        }
    } else {
        println("OK")
    }
}

File: tests/expected/sys_argv.cpp
#include <cassert>   // NOLINT(build/include_order)
#include <cstdint>   // NOLINT(build/include_order)
#include <iostream>  // NOLINT(build/include_order)
#include <string>    // NOLINT(build/include_order)
#include <vector>    // NOLINT(build/include_order)

int main(int argc, char** argv) {
  std::vector<std::string> a = std::vector<std::string>(argv, argv + argc);
  std::string cmd = a[0];
  if (cmd == std::string{"dart"}) {
    /* pass */
  } else {
    assert(cmd.find(std::string{"sys_argv"}) != std::string::npos);
  }
  if (static_cast<int>(a.size()) > 1) {
    std::cout << a[1];
    std::cout << std::endl;
  } else {
    std::cout << std::string{"OK"};
    std::cout << std::endl;
  }
}
